### PR TITLE
Config: Fix connectionstring for remote_cache in defaults.ini

### LINKED ISSUE #17643: redis cache connstr can't be set through environment variable?
**What happened**:

I attempted to use the `redis` backend for `remote_cache`.

To configure it, I exported the environment variables

    GF_REMOTE_CACHE_TYPE=redis
    GF_REMOTE_CACHE_CONNSTR=addr=a.b.c.d:25224,db=0 

Grafana fails to start with the log

> "Service init failed: incorrect redis connection string format detected for '', format is key=value,key=value"

If I set the relevant property in `grafana.ini`, everything works fine; the bug is only when I use the environment variable to attempt to configure the remote_cache subsystem.

**What you expected to happen**:

Grafana starts up and uses the redis instance on host a.b.c.d port 25224

**How to reproduce it (as minimally and precisely as possible)**:

Attempt to configure the remote_cache with the `GF_REMOTE_CACHE_CONNSTR` environment variable

**Anything else we need to know?**:

This is almost certainly because the `connstr` key is not present in `conf/defaults.ini` so cannot be overridden by environment variables.

This is moderately infuriating.

**Environment**:
- Grafana version: 6.2.4
- Data source type & version: N/A
- OS Grafana is installed on: Linux
- User OS & Browser: N/A
- Grafana plugins: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by kylebrandt**
fixes #17643

### CODE REVIEW

**Review by torkelo — APPROVED** [review]

File: pkg/setting/setting_test.go [hunk 1]
```diff
@@ -30,6 +32,21 @@ func TestLoadingSettings(t *testing.T) {
 			So(cfg.RendererCallbackUrl, ShouldEqual, "http://localhost:3000/")
 		})
 
+		Convey("default.ini should have no semi-colon commented entries", func() {
+			file, err := os.Open("../../conf/defaults.ini")
+			if err != nil {
+				t.Error("failed to load defaults.ini file")
```
Comment (kylebrandt) [inline pkg/setting/setting_test.go]: should also include value of err.

**Review by marefr — APPROVED** [review]
Awesome stuff @kylebrandt :+1: Extremely valuable to have test for this! Just fix your comment and you can merge this.

File: pkg/setting/setting_test.go [hunk 2]
```diff
@@ -30,6 +32,21 @@ func TestLoadingSettings(t *testing.T) {
...
+			}
+			defer file.Close()
+
+			scanner := bufio.NewScanner(file)
+			for scanner.Scan() {
+				if strings.HasPrefix(scanner.Text(), ";") {
```
Comment (aocenas) [inline pkg/setting/setting_test.go]: Is it just out convention to comment entries with `;` and comments with `#` or is that in the ini format. It seems to be working both ways for me.

Comment (kylebrandt) [reply pkg/setting/setting_test.go]: I think just a convention. So commenting out the value with # would not be caught. But then I would have to somehow distinguish between a line with `#` that is supposed to be a comment - but maybe there is a reasonable way to do that using the iniFile structure and with text scanning?

Comment (aocenas) [reply pkg/setting/setting_test.go]: Yeah no idea, I think it is not a big issue and no need to handle it, would probably just add a comment that it assumes that only values are commented with `;` and not with `#`.

**Pull request merged**